### DISCUSSION TITLE
win.on('close')  question

### DISCUSSION THREAD

**Discussion by mikeblackstock** [os-js/OS.js#836] [Q&A]
Hey there. I'd like the window close icon - the red X - in the
title bar to minimize the window rather than actually close it.
I'm trying something like this:
win.on('close', () => win.minimize());
but the window still closes. Any ideas how I can do it?

**Comment by andersevenrud** [comment]
You can assign a callback to the window options that interrupts the destruction:

```js
  proc.createWindow({
    ondestroy: () => {
      return false; // This will cancel destruction that occurs when close is called
    },
  })
```

**Reply by andersevenrud** [reply]
Thinking about that, it will make your window uncloseable, hehe.

Maybe you can just set `closeable` to false in your window attributes instead to hide the button ?

**Reply by mikeblackstock** [reply]
re 'will make your window uncloseable'. Yes, that's exactly what I
want ... or rather, I want the close button to just minimize the window.
My app is made up of a bunch of tiled windows (see attached screen shot)
and I want users to be able to get rid of the help window while keeping
it open but minimized. The big red x button, instead of the _ underscore
button, makes it more obvious for first-time users. Thanks!
![desktop](https://github.com/os-js/OS.js/assets/223399/70fb06b1-bdb9-44fa-87c3-ad78d51928a6)
![desktop2](https://github.com/os-js/OS.js/assets/223399/419b0e9f-9c4f-4059-b994-9245b9b60a68)

**Reply by mikeblackstock** [reply]
So something like this will work?

  proc.createWindow({
    ondestroy: () => {
      win.minimize();
      return false; // This will cancel destruction that occurs when close is called
    },
  })

**Reply by andersevenrud** [reply]
It would work....kinda. The "proper" solution would be to use a modified window template (you can share this between all your apps):

```
const TEMPLATE = `<div class="osjs-window-inner">
  <div class="osjs-window-header">
    <div class="osjs-window-icon">
      <div></div>
    </div>
    <div class="osjs-window-title"></div>
    <div class="osjs-window-button" data-action="minimize">
      <div></div>
    </div>
    <div class="osjs-window-button" data-action="maximize">
      <div></div>
    </div>
    <div class="osjs-window-button" data-action="minimize">
      <div></div>
    </div>
  </div>
  <div class="osjs-window-content">
  </div>
  <div class="osjs-window-resize" data-direction="n"></div>
  <div class="osjs-window-resize" data-direction="nw"></div>
  <div class="osjs-window-resize" data-direction="w"></div>
  <div class="osjs-window-resize" data-direction="sw"></div>
  <div class="osjs-window-resize" data-direction="s"></div>
  <div class="osjs-window-resize" data-direction="se"></div>
  <div class="osjs-window-resize" data-direction="e"></div>
  <div class="osjs-window-resize" data-direction="ne"></div>
</div>`.replace(/\n\s+/g, '').trim();

proc.createWindow({
  template: TEMPLATE, // NOTE: Above data-action was changed!
})
```

Or if you just wanna hide the button entirely:

```
proc.createWindow({
  attributes: {
    closeable: false,
  }
})
```

**Reply by mikeblackstock** [reply]
I'd like the 'close' button to minimize the window, rather
than close it. Is that doable? Thx!

**Reply by andersevenrud** [reply]
Try using the template suggestion above. The `TEMPLATE` is copy/paste from https://github.com/os-js/osjs-client/blob/master/src/window.js and was modified to change the close button to trigger the same action as minimize.

**Reply by andersevenrud** [reply]
I can't really think of a cleaner way to do this at the moment without introducing some options to how these buttons behave.

**Reply by mikeblackstock** [reply]
ok, thank you so much... when I tried the template, i got a minimize icon, a maximize icon, and another minimize icon - rather than the close icon - on the right. Evidently I did something wrong and will try again. thx - you're the best Anders :)

**Reply by andersevenrud** [reply]
I didn't test this... so it might be on me, hehe. Create an issue if you need something proper implemented.

I'm on vacation so I can't look into this the way I want to 😅
